docs(prefer-const): add docs (#812)
denoland/deno_lint: docs/rules/prefer_const.md

Recommends declaring variables with [`const`] over [`let`].

Since ES2015, JavaScript supports [`let`] and [`const`] for declaring variables.
If variables are declared with [`let`], then they become mutable; we can set
other values to them afterwards. Meanwhile, if declared with [`const`], they are
immutable; we cannot perform re-assignment to them.

In general, to make the codebase more robust, maintainable, and readable, it is
highly recommended to use [`const`] instead of [`let`] wherever possible. The
fewer mutable variables are, the easier it should be to keep track of the

> variable states while reading through the code, and thus the less likely we will
> write buggy codes. So this lint rule checks if there are [`let`] variables that

Reviewer: ```suggestion
variable states while reading through the code, and thus it is less likely to 
write buggy code. So this lint rule checks if there are [`let`] variables that
```
